middleware/grpc/shared.go

```diff
@@ -1,15 +1,59 @@
-// +build go1.9
-
 package grpc
 
 import (
+	"context"
 	"strings"
 
+	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/stats"
+	"google.golang.org/grpc/status"
+
+	"github.com/openzipkin/zipkin-go"
+	"github.com/openzipkin/zipkin-go/model"
 )
 
+// A RPCHandler can be registered using WithClientRPCHandler or WithServerRPCHandler to intercept calls to HandleRPC of
+// a handler for additional span customization.
+type RPCHandler func(span zipkin.Span, rpcStats stats.RPCStats)
+
 func spanName(rti *stats.RPCTagInfo) string {
 	name := strings.TrimPrefix(rti.FullMethodName, "/")
 	name = strings.Replace(name, "/", ".", -1)
 	return name
 }
+
+func handleRPC(span zipkin.Span, rs stats.RPCStats, handlers []RPCHandler) {
```

Comment:
as mentioned above... replace `span zipkin.Span` for `ctx context.Context` and add logic to pull span from context here.